cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -254,7 +293,7 @@ func (cm *certManagerClient) deleteObjs(objs []unstructured.Unstructured) error
 	return nil
 }
 
-func shouldUpgrade(objs []unstructured.Unstructured) (string, bool, error) {
+func shouldUpgrade(cm *certManagerClient, objs []unstructured.Unstructured) (string, bool, error) {
```

Comment:
what about 
```suggestion
func (cm *certManagerClient) shouldUpgrade(objs []unstructured.Unstructured) (string, bool, error) {
```

Reply:
this is not related to the PR,but I can add as well